elvish-mode.el

```diff
@@ -61,9 +61,12 @@ determined by the syntax table. This allows us to keep things like '-'
 in the symbol part of the syntax table, so `forward-word' works as
 expected.")
 
+(defconst elvish-start-of-statement '(sequence (or line-start "(" ";" "|") (zero-or-more space))
+  "Regex to match the beginning of an Elvish statement (where a command or keyword can appear)")
+
 (defconst elvish-keyword-pattern
   (let ((keywords (cons 'or elvish-keywords)))
-    (eval `(rx symbol-start (group ,keywords) symbol-end)))
+    (eval `(rx ,elvish-start-of-statement (group ,keywords) symbol-end)))
```

Comment:
Using this for keywords breaks highlighting on things like:

    if (== $foo $bar) {
        ...
    } else {    # the else is not highlighted
        ...
    }

You could probably add '}' to the `elvish-start-of-statement` definition, but I don't think I see why a keyword should have to occur after a statement start. That may be generally true right now, but I would think keywords could appear anywhere in principle. Do you have a case where keyword highlighting is wrong with the current pattern?